### ISSUE TITLE
Publish on Flathub

### ISSUE THREAD

**Issue by spazziale** [brandonp2412/Flexify#119]
**Is your feature request related to a problem? Please describe.**
https://flatpak.org/
Even if a tarball is technically functional, having a proper package that can be installed universally in all linux distros is more convenient. In addition to user benefits such as the sandbox and autoupdates, Flexify can gain more visibility as flathub recently surpassed [One million active users ](https://docs.flathub.org/blog/over-one-million-active-users-and-growing/)

**Describe the solution you'd like**
If you want, I can help you build a Flatpak manifest to submit to Flathub

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by spazziale** [comment]
This basic manifest works, it needs the beta version of the sdk due to some incompatibility with glib. Now it only need a `.desktop` and a `.metainfo.xml`. It would be nice to also support aarch64 but I guess that is a separate issue.
`{
    "app-id" : "io.github.brandonp2412.Flexify",
    "runtime" : "org.freedesktop.Platform",
    "runtime-version" : "24.08beta",
    "sdk" : "org.freedesktop.Sdk",
    "command" : "flexify",
    "separate-locales" : false,
    "finish-args" : [
        "--share=ipc",
        "--socket=fallback-x11",
        "--socket=wayland",
        "--socket=pulseaudio",
        "--device=dri"
    ],
    "modules" : [
        {
            "name" : "flexify",
            "buildsystem" : "simple",
            "build-commands": [
            	"mv flexify-linux /app/flexify",
            	"install -d /app/bin/",
		"ln -s /app/flexify/flexify /app/bin/flexify",
		"install -Dm644 /app/flexify/data/flutter_assets/assets/ic_launcher.png /app/share/icons/hicolor/512x512/apps/io.github.brandonp2412.Flexify.png"
            ],
            "sources" : [
                {
                    "type" : "archive",
                    "only-arches" : [
                        "x86_64"
                    ],
                    "url" : "https://github.com/brandonp2412/Flexify/releases/download/1.1.86/flexify-linux.zip",
                    "sha256" : "9183fdf8bf6978b859fb7d999323024e2c382b7f1ffc12bac5df0754e52203fa",
                    "strip-components": 0,
                    "dest": "flexify-linux"
                }
            ]
        }
    ]
}
`

**Comment by brandonp2412** [comment]
Yeah I tried to do this when I first published the app on f-droid but had some errors. 

If your one works as is then I'm happy to accept any PR for it. 

I don't personally like flathub or snap packages I prefer system packages. Not against flexify supporting it though.

**Comment by nico359** [comment]
I also stumbled across this recently after I got a Linux phone. I am using the flutter build bundle now with a desktop shortcut and it works well but a flatpak would of course be more elegant and easier to discover. I tried to figure out how to build a flatpak manifest unsing the flatpak-flutter project because using precompiled binaries is apparently against flathub policies. Please excuse me if there are any issues with how I did it but I am still new to this packaging thing. 

I uploaded the manifest to a github repo and it should hopefully build without any issues now.
https://github.com/nico359/flexify-flatpak/tree/main

The metainfo.xml file is more or less just an example for now. Maybe you could let me know if your still interested in publishing your app as a flatpak.
